app/models/ccms/submission.rb

```diff
@@ -10,7 +10,7 @@ class Submission < ApplicationRecord
     validates :legal_aid_application_id, presence: true
     # rubocop:enable Rails/RedundantPresenceValidationOnBelongsTo
 
-    after_save do
+    after_commit do
```

Comment:
Small matter that arose while googling differences between after_save and after_commit:

> Also, after_commit will execute even if the record was only touched. This may NOT be what you want. after_save will not run after touch.

perhaps we should be used an `on: :create` here too? maybe @colinbruce will know

Reply:
Yeah, where these replace `after_save` they should be `after_commit on: [:create, :update]`. `after_commit` callbacks are run after `destroy` too.